File: day-01/solution.go
package main

import (
	commons_io "commons"
	"strconv"
)

func main() {
	lines := commons_io.ReadLines("day-01/input.txt")
	nums := make([]int, len(lines))
	for i, line := range lines {
		n, err := strconv.Atoi(line)
		if err != nil {
			panic(err)
		}
		nums[i] = n
	}

	threesum := make([]int, len(nums)-2)
	for i := 0; i < len(nums)-2; i++ {
		threesum[i] = nums[i] + nums[i+1] + nums[i+2]
	}
	count := 0
	for i := 1; i < len(threesum); i++ {
		if threesum[i] > threesum[i-1] {
			count++
		}
	}
	println(count)
}

File: day-01/solution.rs
// cargo-deps: commons = { path = "../commons" }
fn main() {
    let lines = io::read_lines("day-01/input.txt");
    let nums = lines
        .unwrap()
        .map(|line| line.unwrap().parse::<i32>().unwrap())
        .collect::<Vec<i32>>();
    let mut threesums: Vec<i32> = vec![0; nums.len() - 2];
    for i in 0..threesums.len() {
        threesums[i] = nums[i] + nums[i + 1] + nums[i + 2];
    }
    let mut count = 0;
    for i in 1..threesums.len() {
        if threesums[i] > threesums[i - 1] {
            count += 1;
        }
    }
    println!("{}", count);
}
